feat(traefik): Add stripPrefixRegex

stdedos opened the pull request:
Signed-off-by: Stavros Ntentos <133706+stdedos@users.noreply.github.com>

**Description**
<!--
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
-->
⚒️ Fixes  # <!--(issue)-->

**⚙️ Type of change**

- [x] ⚙️ Feature/App addition
- [ ] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**
<!--
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
-->

_It hasn't, but then again I was instructed to just open a PR for it https://discord.com/channels/830763548678291466/886351979306188811/960460336572551168_

**📃 Notes:**
<!-- Please enter any other relevant information here -->

**✔️ Checklist:**

- [x] ⚖️ My code follows the style guidelines of this project
- [x] 👀 I have performed a self-review of my own code
- [x] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [x] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [x] ⬆️ I increased versions for any altered app according to semantic versioning

stdedos committed:
feat(traefic): Add stripPrefixRegex

Signed-off-by: Stavros Ntentos <133706+stdedos@users.noreply.github.com>

Crow-Control commented on charts/core/traefik/templates/middlewares/stripPrefixRegex.yaml:
```diff
@@ -0,0 +1,11 @@
+{{ range $index, $middlewareData := .Values.middlewares.stripPrefixRegex }}
+---
+apiVersion: traefik.containo.us/v1alpha1
+kind: Middleware
+metadata:
+  name: {{ $middlewareData.name }}
+  namespace: {{ $namespace }}
+spec:
+  stripPrefixRegex:
+    regex: {{ $middlewareData.regex | quote }}
+{{ end }}
```
Why did you not-copy the whole important section above this?

```suggestion
{{- $values := .Values }}
{{- $namespace := ( printf "ix-%s" .Release.Name ) }}
{{- if or ( not .Values.ingressClass.enabled ) ( and ( .Values.ingressClass.enabled ) ( .Values.ingressClass.isDefaultClass ) ) }}
{{- $namespace = "default" }}
{{- end }}

{{ range $index, $middlewareData := .Values.middlewares.stripPrefixRegex }}
---
apiVersion: traefik.containo.us/v1alpha1
kind: Middleware
metadata:
  name: {{ $middlewareData.name }}
  namespace: {{ $namespace }}
spec:
  stripPrefixRegex:
    regex: {{ $middlewareData.regex | quote }}
{{ end }}
```

stdedos replied:
Because I read the first commit that added the `charts/core/traefik/templates/middlewares/forwardauth.yaml` file 😅

Commit:
Update charts/core/traefik/templates/middlewares/stripPrefixRegex.yaml

Crow-Control reviewed (changes requested):
I'm missing the required minor version bump in Chart.yaml and the definition + comments in values.yaml

stdedos committed:
feat(traefik): Add stripPrefixRegex: Part 2

> I'm missing the required minor version bump in Chart.yaml and the definition + comments in values.yaml

Additionally, add the changelog entry to-be

Signed-off-by: Stavros Ntentos <133706+stdedos@users.noreply.github.com>

Crow-Control commented on charts/core/traefik/CHANGELOG.md:
```diff
@@ -1,6 +1,15 @@
+* Add `stripPrefixRegex` Traefik middleware ([#2422](https://github.com/truecharts/apps/pull/2422), [documentation](https://doc.traefik.io/traefik/middlewares/http/stripprefixregex/))
+
+
+
```
NEVER touch the changelog.
EVER.

All md files are autogenerated by ci touching them breaks things.

Crow-Control commented on charts/core/traefik/Chart.yaml:
```diff
@@ -22,7 +22,7 @@ sources:
 - https://traefik.io/
 type: application
-version: 11.0.13
+version: 11.0.14
```
As requested, please bump the *minor* version, as this is a feature addition not a patch.

Crow-Control reviewed (changes requested):
see above comments

stdedos committed:
feat(traefik): Add stripPrefixRegex: Oops

Signed-off-by: Stavros Ntentos <133706+stdedos@users.noreply.github.com>

stdedos commented:
Oops 😓 

Hopefully it's clean now 🙏

Crow-Control reviewed (approved):
Looks fine to me! :)
Thank you for your submission, may users inter-alia requested this.

Your Helm templating skills are quite good though :)

@all-contributors please add @stdedos for code

stavros-k commented:
@all-contributors please add @stdedos for code

stdedos commented:
> Looks fine to me! :) Thank you for your submission, may users inter-alia requested this.

It took me around 6h-8h bootstrapping (traefik, repo, instructions, piecing together etc), I am glad it was **that** easy at the end to merge!

> Your Helm templating skills are quite good though :)

No, my investigation skills are good enough; my goofness too 😛
Helm has nothing to do with it.

e.g. I have no idea why/how I missed your first comment; nor "what does it mean" anyway.

> @all-contributors please add @stdedos for code

Any "plans" when is this going live?

Crow-Control commented:
Well, I rather have fine copy-pasters than mediocre coders ;-)

New version should be online in a few minutes or so :)
Be sure to bash that refresh button on SCALE :)

truecharts-admin commented:
This PR is locked to prevent necro-posting on closed PRs. Please create a issue or contact staff on discord if you want to further discuss this
